server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1602,6 +1602,8 @@ private SnapshotsInProgress updatedSnapshotsInProgress(ClusterState currentState
             boolean changed = false;
 
             final String repoName = deleteEntry.repository();
+            // Computing the new assignments can be quite costly, only do it once below if actually needed
+            ImmutableOpenMap<ShardId, ShardSnapshotStatus> shardAssignments = null;
```

Comment:
This whole logic could be made more efficient I suppose by computing the reassignments differently (compared to computing them the first time around), but for now this is an easy change that saves a ton of cycles if there's multiple snapshots running concurrently for a repo.